File: systems/Ace-Systems/examples/Quantum-APL-main/tests/test_mu_override_invariants.js
// INTEGRITY_METADATA
// Date: 2025-12-23
// Status: JUSTIFIED - Test file validates system behavior
// Severity: LOW RISK
// Risk Types: ['test_coverage']
// File: systems/Ace-Systems/examples/Quantum-APL-main/tests/test_mu_override_invariants.js

const assert = (c, m) => { if (!c) throw new Error(m || 'assert'); };
const { spawnSync } = require('node:child_process');

// With QAPL_MU_P override, barrier_eq_phi_inv should be false in invariants()
const code = `
  process.env.QAPL_MU_P = '0.600';
  const C = require('./src/constants');
  const inv = C.invariants();
  process.stdout.write(JSON.stringify(inv));
`;

const out = spawnSync(process.execPath, ['-e', code], { env: { ...process.env } });
if (out.status !== 0) throw new Error(out.stderr.toString());
const inv = JSON.parse(out.stdout.toString());
assert(inv.barrier_eq_phi_inv === false, 'barrier_eq_phi_inv should be false when MU_P overridden');
console.log('MU override invariants test passed');

File: systems/Ace-Systems/examples/Quantum-APL-main/tests/test_mu_override_invariants.py
# INTEGRITY_METADATA
# Date: 2025-12-23
# Status: JUSTIFIED - Test file validates system behavior
# Severity: LOW RISK
# Risk Types: ['test_coverage']
# File: systems/Ace-Systems/examples/Quantum-APL-main/tests/test_mu_override_invariants.py

import os
from quantum_apl_python.analyzer import QuantumAnalyzer


def test_analyzer_barrier_override_prints_delta(monkeypatch):
    monkeypatch.setenv('QAPL_MU_P', '0.600')
    results = {
        'quantum': {'z': 0.5, 'phi': 0.0, 'entropy': 0.0, 'purity': 1.0},
        'classical': {},
        'history': {},
        'analytics': {}
    }
    az = QuantumAnalyzer(results)
    text = az.summary()
    # When MU_P is overridden off exact, barrier line shows Δ=...
    assert '(Δ=' in text or 'Delta=' in text

